### PR TITLE
allow SDK to obtain credentials from env vars in lambda/ec2

### PR DESCRIPTION
**Pull request by rschick**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -32,11 +26,7 @@ module.exports = function S3Router(options) {
     s3Options.endpoint = options.endpoint;
   }
 
-  const s3 = new AWS.S3(Object.assign({
```
Comment (rschick) [inline index.js:35]: Let me know if I'm breaking some use case here. I think this is all you need to do.

File: index.js [hunk 2]
```diff
@@ -4,15 +4,9 @@ const Router = require('koa-router');
 const uuid = require('node-uuid');
 const AWS = require('aws-sdk');
 
-const S3_ACCESS_KEY = process.env.AWS_ACCESS_KEY_ID;
-const S3_SECRET_ACCESS_KEY = process.env.AWS_SECRET_ACCESS_KEY;
```
Comment (ktonon) [inline index.js:8]: I don't see these being used at all anymore. The README still says setting environment variables is an option. I could merge, but I would update the README, then I'll do a major version bump

Comment (ktonon) [reply index.js:8]: I guess the AWS SDK picks them up anyhow

Comment (rschick) [reply index.js:8]: I'll update the README.

Comment (rschick) [reply index.js:8]: Actually I think the README is ok. You can still set those environment variables, it's just not recommended in EC2/Lambda. Still useful in dev/test environments though.

Comment (rschick) [reply index.js:8]: Ok I reworded it slightly to hopefully be more clear.

**Review by ktonon — APPROVED** [review]

**Comment by ktonon** [comment]
Published as `1.0.4`

**Pull request merged**